scripts/git-linear-history-check.sh

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bash
+
+# This script is designed to run against a merge commit, either produced during PR checks or a
+# `bors` octopus merge commit. A non-zero exit status is returned if either of the following is true
+# of the COMMITISH (see below)
+# * If COMMITISH is not a merge commit (i.e.: has 0 or 1 parents.)
+# * If COMMITISH is a merge commit and any of the parents have a non-linear history with the common ancestor
+#   of the target commit (first parent).
+
+# usage: ./scripts/git-linear-history-check.sh [COMMITISH]
+#
+#   COMMITISH: a commit to use, if not provided, HEAD is used.
+#
+
+# ## Non-linear merge example
+#
+# Merge commit hash: fc7c341c38c006f96ac288ebfcf5ce18b8e31a48
+#
+# PR commits: https://github.com/pulumi/pulumi/pull/11095/commits
+#
+# This commit is a single merge commit produced by `bors`. The 2nd parent of the commit is the PR
+# commit HEAD, and the commit history shos . This script should log errors and exit with a 1 on this
```

Comment:
"shos ."?

Reply:
Weird! I used an extension to rewrap this after typing it up...